### PR TITLE
[WIP] Add pgbouncer support

### PR DESCRIPTION
**Pull request by andriisoldatenko**
TODO:
- [x] update prisma
- [x] update connection for knex migrations
- [x] update connection to create airflow deployment dbs
- [x] implement new pgbouncer-bootsrapper https://github.com/astronomer/db-bootstrapper/pull/9

fix https://github.com/astronomer/issues/issues/647

### CODE REVIEW

**Comment by schnie** [comment]
@andriisoldatenko were you actually able to test this config against an live db with SSL enabled?

**Comment by andriisoldatenko** [comment]
@schnie i've tested locally and it works:
```
psql -h localhost -p 6543 -d postgres -U postgres
Password for user postgres:
psql (12.1)
Type "help" for help.
postgres=# select * from pg_stat_ssl where pid = pg_backend_pid();
 pid | ssl | version |           cipher            | bits | compression | client_dn | client_serial | issuer_dn
-----+-----+---------+-----------------------------+------+-------------+-----------+---------------+-----------
  34 | t   | TLSv1.2 | ECDHE-RSA-AES256-GCM-SHA384 |  256 | f           |           |               |
(1 row)
postgres=#
1:01
[databases]
postgres = host=postgres dbname=postgres port=5432
[pgbouncer]
listen_port = 6543
listen_addr = *
auth_type = md5
auth_file = /etc/pgbouncer/users.txt
admin_users = postgres
server_tls_sslmode = verify-ca
server_tls_ca_file = /etc/pgbouncer/ssl/server.crt
```

File: charts/astronomer/templates/_helpers.yaml [hunk 1]
```diff
@@ -254,3 +254,19 @@ azure:
...
+{{ printf "%s-pgbouncer-config" .Release.Name }}
+{{- end }}
+
+{{ define "pgbouncer_stats_secret" -}}
+{{ .Release.Name }}-pgbouncer-stats
+{{- end }}
+
+{{ define "pgbouncer_users" }}
+{{ .Values.pgbouncer.connection.user | quote }} {{ .Values.pgbouncer.connection.pass | quote }}
+{{- end }}
```
Comment (schnie) [inline charts/astronomer/templates/_helpers.yaml]:
```suggestion
{{- end }}

```

File: charts/astronomer/values.yaml [hunk 2]
```diff
@@ -159,7 +164,42 @@ prisma:
...
+    # Enabled network policies
+    enabled: false
 
+  connection:
+    user: postgres
+    pass: postgres
+    host: astronomer-postgresql
+    port: 5432
+    db: postgres
+    sslmode: verify-ca
```
Comment (schnie) [inline charts/astronomer/values.yaml]:
```suggestion
    sslmode: verify-ca

```

File: charts/astronomer/values.yaml [hunk 3]
```diff
@@ -25,8 +25,8 @@ images:
     tag: latest
     pullPolicy: Always
   dbBootstrapper:
-    repository: astronomerinc/ap-db-bootstrapper
-    tag: 0.11.0
+    repository: ap-db-bootstrapper
```
Comment (ashb) [inline charts/astronomer/values.yaml]:
```suggestion
    repository: astronomerinc/ap-db-bootstrapper
```

I think you didn't want this change commited.

File: charts/astronomer/templates/houston/houston-backend-secret.yaml [hunk 4]
```diff
@@ -13,9 +13,11 @@ metadata:
     component: {{ template "houston.backendSecret" . }}
 type: Opaque
 data:
-  {{- if .Values.houston.backendSecretConnection }}
+  {{- if .Values.pgbouncer.enabled -}}
+  connection: {{ (printf "postgresql://%s:%s@%s-pgbouncer:%s/%s" .Values.pgbouncer.connection.user .Values.pgbouncer.connection.pass .Release.Name (.Values.ports.pgbouncer | toString) .Values.pgbouncer.connection.db) | b64enc | quote }}
```
Comment (ashb) [inline charts/astronomer/templates/houston/houston-backend-secret.yaml]: It's probably not great to require any passwords in the Values file to enable SSL.

**Comment by andriisoldatenko** [comment]
@sjmiller609  @ashb @schnie PTAL

**Comment by sjmiller609** [comment]
this PR is required first, right? https://github.com/astronomer/db-bootstrapper/pull/9

**Comment by sjmiller609** [comment]
I think we discussed in standup today that some changes may occur due to prisma 2 integration so I'm marking this WIP. feel free to hit me up

**Comment by sjmiller609** [comment]
Is this deprecated? @andriisoldatenko

**Comment by danielhoherd** [comment]
Closing stale PR.

**Pull request closed without merging**